src/keys/slip10.rs

```
@@ -1,32 +1,87 @@
-// Copyright 2020-2021 IOTA Stiftung
+// Copyright 2020-2023 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
 #![allow(clippy::from_over_into)]
 
 use alloc::vec::Vec;
-use core::{convert::TryFrom, default::Default};
+use core::convert::TryFrom;
 
 use serde::{Deserialize, Serialize};
 use zeroize::{Zeroize, ZeroizeOnDrop};
 
-use crate::{macs::hmac::HMAC_SHA512, signatures::ed25519::SecretKey};
+use crate::macs::hmac::HMAC_SHA512;
+#[cfg(feature = "ed25519")]
+use crate::signatures::ed25519;
+#[cfg(feature = "secp256k1")]
+use crate::signatures::secp256k1_ecdsa;
 
 // https://github.com/satoshilabs/slips/blob/master/slip-0010.md
 // https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
 // https://en.bitcoin.it/wiki/BIP_0039
 
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
 pub enum Curve {
+    #[cfg(feature = "ed25519")]
     Ed25519,
+    #[cfg(feature = "secp256k1")]
+    Secp256k1,
 }
 
 impl Curve {
+    pub fn is_non_hardened_supported(&self) -> bool {
+        match self {
+            #[cfg(feature = "ed25519")]
+            Curve::Ed25519 => false,
+            #[cfg(feature = "secp256k1")]
+            _ => true,
```

I think I would explicitly mention `Curve::Secp256k1`, because if we ever add a new curve type, it will silently return true even though it may be false. Or is there really just `ed25519` that can't support it?

---

SLIP-10 applies to 3 curves (ed25519, secp256k1, nist p-256). Only ed25519 doesn't support non-hardened key derivation. In general, it's best to explicitly have patterns for Curve constructors, otherwise `#[cfg(...)]` condition would be quite complex and ugly.

---

Done